teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt

```diff
@@ -0,0 +1,216 @@
+package com.mapbox.vision.teaser.ar
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.annotations.Marker
+import com.mapbox.mapboxsdk.annotations.MarkerOptions
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.location.LocationComponent
+import com.mapbox.mapboxsdk.location.LocationComponentActivationOptions
+import com.mapbox.mapboxsdk.location.LocationComponentOptions
+import com.mapbox.mapboxsdk.location.modes.CameraMode
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.services.android.navigation.ui.v5.route.NavigationMapRoute
+import com.mapbox.services.android.navigation.v5.navigation.NavigationRoute
+import com.mapbox.vision.teaser.R
+import com.mapbox.vision.utils.VisionLogger
+import kotlinx.android.synthetic.main.activity_ar_map.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class ArMapActivity : AppCompatActivity(), MapboxMap.OnMapClickListener, OnMapReadyCallback {
+
+    companion object {
+        private var TAG = ArMapActivity::class.java.simpleName
+        private const val MAP_STYLE = "mapbox://styles/mapbox/dark-v10"
+    }
+
+    private var originPoint: Point? = null
+
+    private lateinit var mapboxMap: MapboxMap
+
+    private var destinationMarker: Marker? = null
+
+    private var currentRoute: DirectionsRoute? = null
+    private var navigationMapRoute: NavigationMapRoute? = null
+    private var locationComponent: LocationComponent? = null
+
+    private val arLocationEngine by lazy {
+        LocationEngineProvider.getBestLocationEngine(this)
+    }
+
+    private val arLocationEngineRequest by lazy {
+        LocationEngineRequest.Builder(0)
```

Comment:
maybe add name for parameter? or it's from Java?

Reply:
yeah, java mabox-core-sdk